### PR TITLE
Features and improvements pack

### PR DESCRIPTION
**Pull request by baturich**
* Close TODOs in Dockerfile (COPY chown)
* Improve performance of concurrency tools (concurrent-ruby-ext gem providing native concurrency C extension)
* Add support to NewRelic Application Performance Monitoring (we prefer this over Scout)
* AMQP: Add ability to configure auto_delete option, queue length and pass custom arguments (at last max queue length is required to make sure RabbitMQ doesn't crash in case lot of messages are being stuck)
* ~~Close TODOs in User API JWT authenticator (use UID instead of email)~~
* Drop unused CSS classes
* Add specs to Operations. build_account_number
* Add hot reload to Workers:: Daemons:: Blockchain (close TODO)
* Enable JS compressor (it was disabled)
* Use optimized JSON parser in Grape (20-50% improvement), allow to customize
* Add specs to Ethereum::Wallet#prepare_deposit_collection (close TODO)
* Fixed few errors in words and few code formatting issues

### CODE REVIEW

**Comment by baturich** [comment]
Hey there

We recently started with Peatio deployment and found few things which could be improved.

**Comment by baturich** [comment]
Attaching RSpec logs

```
Randomized with seed 54586
.......................................................................................................................................................................................................*............................................................................................................................................................................................................................................**.*..............****...........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................***......................................................................................................................................................................................................................................................................................................................*.................................................................................................................................................................................................................................................................................................................................

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) Admin::MarketsController#update updates market attributes
     # Temporarily skipped with xit
     # ./spec/controllers/admin/markets_controller_spec.rb:63

  2) Withdraw coin withdraw #audit! should accept withdraw with clean history
     # Temporarily skipped with xit
     # ./spec/models/withdraw_spec.rb:64

  3) Withdraw coin withdraw #audit! should be rejected if address is invalid
     # Temporarily skipped with xit
     # ./spec/models/withdraw_spec.rb:29

  4) Withdraw coin withdraw #audit! sum less than quick withdraw limit should approve quick withdraw directly
     # Temporarily skipped with xit
     # ./spec/models/withdraw_spec.rb:72

  5) Withdraw CashAddr valid CashAddr address
     # Temporarily skipped with xit
     # ./spec/models/withdraw_spec.rb:590

  6) Withdraw CashAddr valid legacy address
     # Temporarily skipped with xit
     # ./spec/models/withdraw_spec.rb:603

  7) Withdraw CashAddr invalid legacy address
     # Temporarily skipped with xit
     # ./spec/models/withdraw_spec.rb:608

  8) Withdraw CashAddr invalid CashAddr address
     # Temporarily skipped with xit
     # ./spec/models/withdraw_spec.rb:595

  9) API::V2::Account::Deposits GET /api/v2/account/deposit_address/:currency successful return cash address
     # Temporarily skipped with xit
     # ./spec/api/v2/account/deposits_spec.rb:149

  10) API::V2::Account::Deposits GET /api/v2/account/deposit_address/:currency successful doesn't expose sensitive data
     # Temporarily skipped with xit
     # ./spec/api/v2/account/deposits_spec.rb:144

  11) API::V2::Account::Deposits GET /api/v2/account/deposit_address/:currency successful return legacy address
     # Temporarily skipped with xit
     # ./spec/api/v2/account/deposits_spec.rb:154

  12) Admin::WalletsController#update updates wallet attributes
     # Temporarily skipped with xit
     # ./spec/controllers/admin/wallets_controller_spec.rb:50


Finished in 18 minutes 21 seconds (files took 2.81 seconds to load)
1648 examples, 0 failures, 12 pending

Randomized with seed 54586```

File: app/api/v2/constraints.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ def included(base)
 
         def apply_rules!
           Rack::Attack.throttle 'Limit number of calls to API', limit: 6000, period: 5.minutes do |req|
-            req.env['api_v2.authentic_member_email']
+            req.env['api_v2.authentic_member_uid']
```
Comment (mod) [inline app/api/v2/constraints.rb]: Could explain this change,
We are moving to a rack middleware I think there is a pull request related

Comment (baturich) [reply app/api/v2/constraints.rb]: @mod I followed this TODO https://github.com/openware/peatio/blob/master/app/api/v2/auth/middleware.rb#L11

Now I see rack-jwt in pull requests. Do you want me to rollback that change?

Comment (baturich) [reply app/api/v2/constraints.rb]: @mod I removed UID change.

**Comment by baturich** [comment]
Hey guys
Any updates? Why it's taking so long to merge small PR?

@mod @ysv @mnaichuk @chumaknadya @calj

**Comment by mod** [comment]
@baturich Hello can you contact me on telegram: t.me/modhelios

**Comment by mod** [comment]
@akulakovaa can you rebase, remove new_relic, oj/grape changes,

File: app/workers/daemons/blockchain.rb [hunk 2]
```diff
@@ -4,7 +4,6 @@
 module Workers
   module Daemons
     class Blockchain < Base
```
Comment (mod) [inline app/workers/daemons/blockchain.rb:6]: remove this change

**Review by mod — CHANGES_REQUESTED** [review]

**Pull request closed without merging**